src/audio/pcm_converter/pcm_converter.c

```diff
@@ -31,12 +31,6 @@ int pcm_convert_as_linear(const struct audio_stream __sparse_cache *source, uint
 	int chunk;
 	int N1, N2;
 
-	/* assert enough avail/free samples in source and sink buffer */
-	if (audio_stream_get_avail_samples(source) < samples + ioffset)
-		return -EINVAL;
-	if (audio_stream_get_free_samples(sink) < samples + ooffset)
-		return -EINVAL;
```

Comment:
an example of where these PCM processing functions are called in https://github.com/thesofproject/sof/blob/8f6018057b64441f26de6d509ac16514d7347f12/src/lib/dma.c#L339 and it seems like there the check isn't complete - it gets available data from the source, but doesn't check whether the sink has sufficient free space. Not sure if this can ever become a problem (e.g. with multiple sources writing data to a single sink), but if that does have to be checked, maybe it has to be checked uniformly for all such calls.

Reply:
@btian1 Can you comment?

Reply:
I like "uniform", sof did a lot of check in many functions, hard to say those check are all must, especially those on critical timing path.

based on current sof architecture(2X period buffer size),  source is checked by samples( as long as it is not zero).

for sink, I am not sure whether there is a case with multi src --> single sink, if there have, single sink buffer size still need 2X of total src, I mean, if sink have no space, there is pipeline overrun.

we can add sink check if you still feel not safe enough.

Reply:
seems I reverted https://github.com/thesofproject/sof/pull/3251 in this patch, @ktrzcinx , may I know any reason to add those checkers in patch #3251

Reply:
@lgirdwood @lyakh @mmaka1 , if you think keep error handling is must for converter, I can remove this patch, and find other way to do optimization.

Reply:
@btian1 before #3251 there already was this check as an `assert()`. The change was related with some bug which ends up with SOF crash so I've changed `assert()` to `if() return` to gain human readable log about root cause of the issue in a similar cases in the future.

Reply:
@ktrzcinx ， this part can be added back, I mean the whole #3251, in this patch, I almost reverted all.